LoRaEngine/LoraKeysManagerFacade/LoRaDeviceCacheRedisStore.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace LoraKeysManagerFacade
+{
+    using System;
+    using Microsoft.Azure.WebJobs;
+    using Microsoft.Extensions.Configuration;
+    using StackExchange.Redis;
+
+    public class LoRaDeviceCacheRedisStore : ILoRaDeviceCacheStore
+    {
+        private IDatabase redisCache;
+
+        public LoRaDeviceCacheRedisStore(ExecutionContext context)
+        {
+            var config = new ConfigurationBuilder()
+                                .SetBasePath(context.FunctionAppDirectory)
+                                .AddJsonFile("local.settings.json", optional: true, reloadOnChange: true)
+                                .AddEnvironmentVariables()
+                                .Build();
+
+            var redisConnectionString = config.GetConnectionString("RedisConnectionString");
+
+            if (string.IsNullOrEmpty(redisConnectionString))
+            {
+                string errorMsg = "Missing RedisConnectionString in settings";
+                throw new Exception(errorMsg);
+            }
+
+            var redis = ConnectionMultiplexer.Connect(redisConnectionString);
+            this.redisCache = redis.GetDatabase();
+        }
+
+        public bool LockTake(string key, string value, TimeSpan timeout)
+        {
+            return this.redisCache.LockTake(key, value, timeout);
```

Comment:
What is your take in using redis async as opposed to sync?

Reply:
Generally, I like to use async where they are provided. I sticked to this because it has been used before and I was assuming the perf characteristics were good enough for this to be a sync call. I think for now, I would keep it this way, unless we see a problem with scalability (which is unlikely in my opinion)